homeassistant/components/sensor/whois.py

```diff
@@ -0,0 +1,126 @@
+"""
+Get WHOIS information for a given host.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.whois/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pythonwhois==2.4.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOST = 'hosts'
+
+ATTR_NAME_SERVERS = 'name_servers'
+ATTR_REGISTRAR = 'registrar'
+ATTR_UPDATED = 'updated'
+ATTR_EXPIRES = 'expires'
+
+SCAN_INTERVAL = timedelta(hours=24)  # WHOIS info is very slow moving
+# We also want to prevent DOS / TOS breaking; One request per domain
+# every 24 hours shouldn't count as "high volume"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): vol.All(cv.ensure_list, [cv.string]),
```

Comment:
Please allow the user to choose a name for the sensor. Hint: `CONF_NAME`.

Reply:
Which style should I go for, Instantiate this sensor with only a single host, such as:
```yaml
sensor:
  - platform: whois
    hostname: example.net
    name: primary
  - platform: whois
    hostname: example.com
    name: secondary
```

or keep the list of hostnames and add the per host name config, such as:
```yaml
sensor:
  - platform: whois
    hostnames:
      - url: example.net
        name: primary
      - url: example.com
        name: secondary
```

I can see merits in both but was curious if there was a process / design used to decide which one to go for.

Thanks for your feedback on this all, much appreciated.

Reply:
I personally prefer the first style (easier for copy-&-paste) but this is pretty much up to the person who is writing the platform.